runtime/src/accounts_db.rs

```diff
@@ -4482,6 +4482,14 @@ impl AccountsDb {
             );
         }
 
+        self.handle_dropped_roots_for_ancient(dropped_roots);
+
+        self.shrink_ancient_stats.report();
+    }
+
+    /// each slot in 'dropped_roots' has been combined into an ancient append vec.
+    /// We are done with the slot now forever.
+    fn handle_dropped_roots_for_ancient(&self, dropped_roots: Vec<Slot>) {
         if !dropped_roots.is_empty() {
```

Comment:
is checking `is_empty` necessary or improves performance? I'd assume rust is smart enough to optimize out any code for iter on an empty iterator.

Reply:
it means I don't have to worry in the rest of the function about side effects of the empty case.